src/common/transformations/src/transformations/common_optimizations/transpose_sinking_unary.cpp

```diff
@@ -0,0 +1,106 @@
+#include "openvino/op/util/op_types.hpp"
+#include "openvino/opsets/opset9.hpp"
+#include "openvino/pass/pattern/op/label.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/util/common_util.hpp"
+#include "openvino/util/log.hpp"
+#include "itt.hpp"
+#include <ngraph/opsets/opset9.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <transformations/utils/utils.hpp>
+
+#include "transformations/common_optimizations/transpose_sinking_unary.hpp"
+
+#include <utility>
+
+namespace {
+
+using NodePtr = std::shared_ptr<ov::Node>;
+using NodePair = std::pair<NodePtr, NodePtr>;
+
+NodePair SwapNodes(NodePtr first_node, NodePtr second_node) {
+    auto second_node_inputs = second_node->input_values();
+    second_node_inputs[0] = first_node->input_value(0);
+    auto new_first_node = second_node->clone_with_new_inputs(second_node_inputs);
+    
+    auto first_node_inputs = first_node->input_values();
+    first_node_inputs[0] = new_first_node;
+    auto new_second_node = first_node->clone_with_new_inputs(first_node_inputs);
+
+    new_second_node->set_friendly_name(second_node->get_friendly_name());
+    ov::copy_runtime_info({first_node, second_node}, {new_first_node, new_second_node});
+
+    ov::replace_node(second_node, new_second_node);
+
+    return std::make_pair(new_first_node, new_second_node);
+}
+
+} // namespace
+
+ov::pass::TransposeSinkingUnaryForward::TransposeSinkingUnaryForward() {
+    MATCHER_SCOPE(TransposeSinkingUnaryForward);
+
+    auto transpose_label =
+        ov::pass::pattern::wrap_type<ov::opset9::Transpose>({ov::pass::pattern::any_input(),
+                                                             ov::pass::pattern::wrap_type<ov::opset9::Constant>()},
+                                                             ov::pass::pattern::consumers_count(1));
```

Comment:
What if we have several consumers? Why are we not going to perform Transpose sinking?

Reply:
It will be covered in the next PR

Reply:
fixed